### ISSUE TITLE
Missing batch dimension of the masks in BatchedVideoDatapoint.

### ISSUE THREAD

**Issue by EricLina** [facebookresearch/sam2#373]
Hello! 
Thanks for your release of the training code. While reading the training code, I have some confusion regarding the data format in BatchedVideoDatapoint.

The dimensions of img_batch and masks do not match[ here](https://github.com/facebookresearch/sam2/blob/c98aa6bea377d5c000cdc80197ce402dbf5304dc/training/utils/data_utils.py#L40C1-L42C88
), 
```
img_batch:  [TxBxCxHxW]
masks:  [TxOxHxW]
```

Ideally, masks should have a batch dimension (corresponding to img_batch). 

Why is it missing here, is this a bug? If not, does this mean that the batch size can only be set to 1 during training?

Your insights and guidance on resolving this issue would be greatly appreciated. Thanks.

**Comment by haithamkhedr** [comment]
Hi @EricLina, the number of masks in a batch of videos is variable depending on the number of objects in each video. We stack all the masks in a batch of videos in the second dimension of the `masks` tensor (denoted by `O`). The batch size (`B` - number of videos) can be arbitrary during training. I hope this resolves the confusion, please let me know if you have further questions.
